### DISCUSSION TITLE
Export and import users of a realm including credentials

### DISCUSSION THREAD

**Discussion by NilsEngelbach** [keycloak/keycloak#25052] [Q&A]
I am trying to export and import all users of a realm including the credentials (hashed passwords).

According to the documentation there are two possible ways:

1. Using [Export and Import](https://www.keycloak.org/server/importExport): `kc.sh export/import`

The exported user files include the credentials but when i try to import only the user files, it says "that the realm already exists", deletes it and than fails at the import because it is not a complete realm file, but only the users.

I don't want to override/import the whole realm, I only want to import the users. Is there a possibility to achieve this?

2. Using `kcadm.sh` or the REST API

Exporting users does not contain the credentials (hashed passwords).

I could try to import the users exported with `kc.sh export` using the API, but this feels a little bit hacky. Is there another option or trick?

**Comment by nmazzon** [comment]
Hi, 
 I am searching a similar options. 

Thanks for your answers.

**Reply by ManiBharathi-R** [reply]
Hi,
Were you able to find the answer if so, Please help me .

Thanks for your answer.

**Comment by lucribas** [comment]
Hi,
Check this:
[https://keepgrowing.in/tools/keycloak-in-docker-5-how-to-export-a-realm-with-users-and-secrets/](keycloak-in-docker-5-how-to-export-a-realm-with-users-and-secrets)
[https://keepgrowing.in/tools/keycloak-in-docker-2-how-to-import-a-keycloak-realm/](keycloak-in-docker-2-how-to-import-a-keycloak-realm)

**Reply by ygotame** [reply]
Don't click the link and copy & paste in different tab, and it's going to work

**Reply by ManiBharathi-R** [reply]
Sorry I'm still not able to open the link. Can you please mention the link again.

**Reply by alexanderankin** [reply]
https://keepgrowing.in/tools/keycloak-in-docker-2-how-to-import-a-keycloak-realm/
https://keepgrowing.in/tools/keycloak-in-docker-5-how-to-export-a-realm-with-users-and-secrets/
https://simonscholz.dev/tutorials/keycloak-realm-export-import
